stac_mjx/io.py

```diff
@@ -178,34 +262,107 @@ def _load_params(param_path):
+        marker_sites (np.ndarray): Marker sites array.
+        offsets (np.ndarray): Offsets array.
+        qpos (np.ndarray): Qpos array.
+        xpos (np.ndarray): Xpos array.
+        xquat (np.ndarray): Xquat array.
+        qvel (np.ndarray): Qvel array.
+        file_path (str): Path to the HDF5 file.
+    """
+    with h5py.File(file_path, "w") as f:
+        # Save config as a YAML string
+        config_yaml = OmegaConf.to_yaml(OmegaConf.structured(config))
+        f.create_dataset("config", data=np.string_(config_yaml))
+
+        # Save stac output data
+        f.create_dataset("kp_names", data=np.array(kp_names, dtype="S"))
+        f.create_dataset("names_qpos", data=np.array(names_qpos, dtype="S"))
+        f.create_dataset("names_xpos", data=np.array(names_xpos, dtype="S"))
+        f.create_dataset("kp_data", data=kp_data, compression="gzip")
+        f.create_dataset("marker_sites", data=marker_sites, compression="gzip")
+        f.create_dataset("offsets", data=offsets, compression="gzip")
+        f.create_dataset("qpos", data=qpos, compression="gzip")
+        f.create_dataset("qvel", data=qvel, compression="gzip")
+        f.create_dataset("xpos", data=xpos, compression="gzip")
+        f.create_dataset("xquat", data=xquat, compression="gzip")
+
+
+def load_stac_data(file_path) -> tuple[Config, StacData]:
+    """Load configuration and STAC data from an HDF5 file.
 
     Args:
-        fit_data (numpy array): Data to write out.
-        save_path (Text): Path to save data. Defaults to None.
+        file_path (str): Path to the HDF5 file.
+
+    Returns:
+        tuple: A tuple containing the Config and StacData dataclasses.
     """
-    if os.path.dirname(save_path) != "":
-        os.makedirs(os.path.dirname(save_path), exist_ok=True)
-    _, file_extension = os.path.splitext(save_path)
-    if file_extension == ".p":
-        with open(save_path, "wb") as output_file:
-            pickle.dump(fit_data, output_file, protocol=2)
-    elif file_extension == ".h5":
-        ioh5.save(save_path, fit_data)
-    else:
-        with open(save_path + ".p", "wb") as output_file:
-            pickle.dump(fit_data, output_file, protocol=2)
+    with h5py.File(file_path, "r") as f:
+        # Load config from YAML string
+        config_yaml = f["config"][()].decode("utf-8")
+        config = OmegaConf.create(config_yaml)
+        config = OmegaConf.structured(Config(**config))
+
+        # Load additional values
+        kp_names = [name.decode("utf-8") for name in f["kp_names"]]
+        names_qpos = [name.decode("utf-8") for name in f["names_qpos"]]
+        names_xpos = [name.decode("utf-8") for name in f["names_xpos"]]
+        kp_data = np.array(f["kp_data"])
+        marker_sites = np.array(f["marker_sites"])
+        offsets = np.array(f["offsets"])
+        qpos = np.array(f["qpos"])
+        qvel = np.array(f["qvel"])
+        xpos = np.array(f["xpos"])
+        xquat = np.array(f["xquat"])
```

Comment:
Canonical way to load the datasets into memory is like:

```py
with h5py.File(file_path, "r") as f:
    kp_data = f["kp_data"][()]
```

Ref: https://docs.h5py.org/en/stable/high/dataset.html#reading-writing-data

Numpy array casting has the same effect by forcing the read op, but doing it the way above maintains type and works with scalars and other primitives without coercing them into arrays.